[지구] - step 5 구현중

Ojeegu opened the pull request:
안녕하세요 Cooper. 이번주 리뷰도 잘 부탁드립니다.
😢 이전에 merge했던 파일을 pull 하지 않고 곧바로 수정해버려서 이 점 양해해주시면 감사하겠습니다..! 😢

<br>

### 💻 URL convention

HTTP Method | URL | 기능
-- | -- | --
GET | / | 게시글 목록 조회
GET | /user/new | 회원가입 폼
POST | /user/new-try | 회원가입
GET | /users | 회원 목록 조회
GET | /users/{id} | 프로필 조회
GET | /users/{id}/revision | 회원 프로필 수정 폼
PUT | /users/{id} | 회원 프로필 수정
GET | /post/new | 게시글 작성 폼
POST | /post/new | 게시글 작성
GET | /posts/{id} | 게시글 상세 조회
GET | /posts/{id}/revision | 게시글 수정 폼
PUT | /posts/{id} | 게시글 수정
GET | /login | 로그인 폼
POST | /login-try | 로그인
POST | /logout | 로그아웃


<br>

### 👀 질문 정리
(1) Article class에 `setter`를 없앴더니 다음과 같은 문제가 생겼습니다.
- 글 작성 후 홈 화면에 글 목록이 제대로 보이지 않음
- 수동으로 `/posts/1` 로 들어가보면 글 양식은 존재하지만 제목, 닉네임, 내용 아무것도 보이지 않고 기본 템플릿만 보임
- 수정 버튼을 누르면 자동으로 `/posts/1/revision`으로 이동해야 하는데 id를 읽지 못하고 `/posts//revision`으로 이동

`setter`를 넣으면 위와 같은 사항이 모두 사라집니다. 심지어 `setCreate_time` 메서드는 사용하는 곳이 없는데도 삭제하면 글 상세보기에서 작성시간이 보이지 않습니다. 이유를 찾아보려 해도 나오지 않는데 혹시 어떤 것을 고쳐야하는 걸까요?
** 해당 현상은 User class에서도 동일하게 나타납니다.

<br>

(2) `toEntity`와 `fromEntity`에 대해 조금 공부해 보았습니다. 아래에서 확인하실 수 있듯, 생성 원리와 `static`/`non-static` 차이에 대해서는 이해가 갔는데 의존성 방향이나 어떤 것이 더 유연한 것인지는 아직 잘 감이 잡히지 않습니다. 이와 관련해서 힌트 주시면 감사하겠습니다.

<br>

(3) 지난 번 피드백 때 공유해주신 [REST API 제대로 알고 사용하기](https://meetup.nhncloud.com/posts/92) 글을 읽어보았습니다. 해당 링크에서는 `DeleteMapping`과 `GetMapping` 각각의 메서드가 `/members/1` 라는 URL을 설정하도록 예시를 보여주고 있습니다.
제가 이전에 URL에 대해 찾아보았을 때는 URL이 고유한 리소스를 가져야 한다고 배웠는데, HTTP method가 다르면 URL이 같아도 상관이 없는 건지 궁금합니다.

<br>

### ✍ 리뷰 피드백 정리
✔ 불필요한 주석 및 사용하지 않는 코드 제거
✔ 네이밍 변경

- `ArticleService.clickOne` → `ArticleService.findById`
- `cehckIdAndPw` → `checkLoginUser`


<br>

✔ `Setter` 대신 메서드 활용
- `matchPassword()` 메서드 추가
- `update()` 메서드 추가


<br>

✔ final 키워드의 장점
(1) final 키워드
- 변수, 메서드, 클래스에 사용
- 변수 : 변수 값 변경 불가능
- 메서드 : 오버라이딩 불가능
- 클래스 : 상속 불가능

(2) 장점
- 해당 변수, 메서드, 클래스가 변경되지 않는 것이 보장되므로 컴파일러가 최적화 작업을 수행 할 수 있다.
- 안정성과 성능 향상에 도움이 된다.

<br>

✔ `fromEntity` vs `toEntity`
``` java
public class JoinDTO { //회원가입용

    private final Long id;
    private final String userId;
    private final String name;
    private final String password;
    private final String email;

    
    public JoinDTO(Long id, String userId, String name, String password, String email) {
        this.id = id;
        this.userId = userId;
        this.name = name;
        this.password = password;
        this.email = email;
    }

    //Entity -> DTO
    public static JoinDTO from(User user) {
        return new JoinDTO(user.getId(), user.getUserId(), user.getName(), user.getPassword(), user.getEmail());
    }

    //DTO -> Entity
    public User toEntity() {
        return new User(userId, name, password, email);
    }

		...

}
```

(1) fromEntity
- Entity 객체의 정보를 기반으로 새로운 DTO 객체를 생성하는 메서드
- DTO 객체를 생성하는 데 필요한 정보가 Entity 객체에 이미 저장되어 있으므로 Entity 객체의 정보만으로 DTO 객체를 생성할 수 있음
- 별도의 DTO 인스턴스 변수에는 접근할 필요가 없음
⇒ static 선언

(2) toEntity
- DTO 객체가 가지고 있는 정보를 활용하여 Entity 객체를 생성하는 메서드
- DTO 인스턴스 변수 값이 필요하므로 DTO 인스턴스 객체를 생성한 뒤 toEntity 메서드를 호출해야 함
- 만약 static 선언하면 DTO 인스턴스 변수에 접근할 수 없으므로 Entity 객체를 생성하는 데 필요한 정보를 제공할 수 없음
⇒ non-static 선언

<br>

✔ 생성자 주입 권장 이유
(1) 불변
- 대부분의 의존관계 주입은 한 번 일어나면 애플리케이션 종료시점까지 의존관계를 변경 할 일이 없으며, 오히려 변경되어선 안된다.
- 생성자 주입은 객체를 생성할 때 딱 1번만 호출되고 이후에 호출되는 일이 없으므로 불변 객체를 설계할 수 있다.

(2) 누락
- 생성자 주입을 사용하면 주입해야 하는 의존성 객체를 누락했을 때 컴파일 오류가 발생하므로 코드의 안정성을 높일 수 있다.

(3) final 키워드
- `final` 키워드를 사용하면 생성자에서 값이 설정되지 않은 오류를 컴파일 시점에서 방지할 수 있다.

<br>

✔ setter 지양 이유
- 사용 의도를 파악하기 힘들다 : `setter`를 사용하여 값을 바꾸는 이유가 최초 생성인지, 수정 변경인지 코드만 봐서는 알 수 없다. 반면, `update(String name)`과 같은 메서드를 사용하면 의도를 한눈에 파악할 수 있다.
- 일관성을 깨뜨린다 : `setter`를 사용하게 되면 어디서든 값을 변경할 수 있게 되므로 객체 일관성이 깨질 수 있다.
- 유지보수가 힘들다 : 객체에 새로운 필드가 추가되면 객체를 생성하는 모든 코드를 찾아 일일이 `setter`를 추가해야 한다.

<br>

✔ session.removeAttribute() 와 session.invalidate() 차이
: 두 메서드 모두 HttpSession 객체의 속성(attribute)를 제거하는 역할을 하지만 제거하는 대상과 방식에 차이가 있다.

(1) session.removeAttribute()
- 세션에서 특정 데이터만 제거하고 다른 속성들은 유지
- 세션은 계속해서 사용 가능

(2) session.invalidate()
- 세션의 모든 데이터 제거
- 세션을 완전히 삭제함으로써 새로운 HttpSession 객체를 생성해야 한다.
- 
⇒ 로그아웃 기능을 구현할 때는 `session.invalidate()` 메서드를 사용해서 로그아웃 한 사용자의 세션 정보를 완전히 삭제하는 것이 좋다.

<br>

✔ @PathVariable 은 값이 존재하지 않을 경우에는 어떻게 동작할까?

(1) `@PathVariable` : Spring MVC에서 사용하는 어노테이션으로, RESTful 웹 서비스의 요청 URL에서 특정 값을  추출하기 위해 사용.
``` arduino
http://example.com/users/123
```

위와 같은 URL이 있다고 할 때 123은 요청한 사용자의 ID를 나타낸다. 이때 `@PathVariable`을 사용하면 다음과 같이 ID값을 추출할 수 있다.

``` java
@GetMapping(&quot;/users/{userId}&quot;)
public User getUserById(@PathVariable Long userId) {
    // userId 값에 따라 해당하는 사용자의 정보를 반환하는 로직 구현
}
```

`@PathVariable`을 사용하면 userId 변수에 URL에서 추출한 값을 자동으로 할당한다.

(2) URL에서 추출할 값을 찾지 못한 경우에는?
: `MissingPathVariableException` 예외를 발생 시킨다. 이 경우 클라이언트에게 경로 변수 값이 누락된 것을 알리기 위해 적절한 에러 메세지를 출력하거나, 클라이언트를 다른 URL로 리다이렉트 할 수 있다.

<br>

### 📝 아직 처리하지 못한 사항
- [ ] 예외 처리 커스텀 하기
- [ ] ControllerAdvice 사용하기
- [ ] 회원가입 시 입력값 유효성 검사 추가
- [ ] 글작성 시 입력값 유효성 검사 추가
- [ ] 5단계 완료하기

<br>

Ojeegu committed:
refactor : 필드명 변경

user 프로필 변경 시 필드 변수를 직관적으로 이해하기 위해 필드명에 new를 추가

Ojeegu committed:
refactor : builder pattern 삭제

builder 패턴 사용 시 잦은 오류가 발생하여 ArticleDTO 클래스의 builder 패턴을 삭제하고 기존의 생성자 패턴으로 수정.

Ojeegu committed:
refactor : 데이터베이스 컬럼에 작성자 아이디, 작성 시간 추가

이전에 작성자로 설정했던 컬럼을 작성자 아이디로 바꾸고 작성 시간을 추가

Ojeegu committed:
feat : 현재 시간 생성 및 문자열 변환 메서드 구현

LocalDateTime class를 이용해 현재 시간을 생성하고 toString을 오버라이딩하여 문자열로 변환하는 기능 추가

Ojeegu committed:
feat : create(), update() 메서드 추가

Article class에 id값과 article 객체를 매개변수로 받는 create() 메서드와 articleDTO를 매개변수로 받는 update() 메서드를 생성하여 setter의 사용을 최소화 함

Ojeegu committed:
feat : 글 작성 시 로그인 한 사용자의 아이디를 매개변수로 넘겨주는 기능 추가

Article class의 userId 값을 설정하기 위해 현재 session에 저장된 loginDTO에서 userId 값을 가져와 Article 매개변수로 전달하는 기능을 구현

Ojeegu committed:
refactor : setter 삭제 및 try-catch문 간결화

setter 대신 메서드를 사용하고, queryForObject 대신 queryForStream을 사용하므로써 코드를 간결화 시킴

Ojeegu committed:
refactor : URL 변경

GetMapping과 PostMapping의 URL 주소를 각기 다르게 설정하고, 동사 사용 자제 및 직관적으로 파악할 수 있도록 수정함.

Ojeegu committed:
feat : 로그인 여부 확인 및 로그인 한 사용자의 id 추출 메서드 추가

userService class에서 로그인 여부를 확인하는 메서드와 로그인 한 사용자의 id를 가져와 매개변수로 전달된 id와 같은지 확인하는 메서드를 구현함.

Ojeegu committed:
feat : 현재 비밀번호와 매개변수로 전달된 비밀번호 동일 여부 확인 메서드 추가

User class에 저장된 현재 비밀번호와 매개변수로 전달된 originPassword가 같은지 확인하고 boolean type을 return하는 메서드 구현

Ojeegu committed:
feat : 수정하려는 프로필 유저와 현재 로그인한 유저가 다를 경우 에러 페이지로 이동하는 기능 추가

수정하려는 프로필의 유저 id와 현재 로그인 한 유저 id를 비교하여 같을 경우에만 수정 페이지로 이동하고, 다를 경우 403 에러 페이지로 이동하여 수정하지 못하도록 구현함.

Ojeegu committed:
refactor : setter 삭제 및 try-catch 간결화

setter 대신 create() 메서드 사용, queryForObject 대신 queryForStream을 사용하여 코드 간결화

Ojeegu committed:
refactor : 메서드 간결화

cehckLoginUser 메서드 사용 시 User class에 추가된 matchPassword 메서드를 이용하여 코드 간결화

Ojeegu committed:
refactor : 필요 없는 메서드 삭제

세션 생성만으로도 충분하므로 쿠키 생성하고 삭제하는 메서드를 삭제함

Ojeegu committed:
feat : 로그인하지 않은 사용자의 글보기 및 글쓰기 기능을 제한

로그인하지 않은 채 글을 보려하거나 글을 쓰려고 할 경우 로그인 페이지로 이동하도록 기능 구현. 로그인 한 경우에만 해당 기능이 활성화 됨.

Ojeegu committed:
refactor : 로직이 필요한 URL 삭제

addViewControllers에는 view를 반환하는 URL만 설정할 수 있으므로 로직이 필요한 /post/write URL 삭제

Ojeegu committed:
feat : 403 에러 페이지 추가

html 파일에 이미지를 추가하여 403 에러 발생 시 웹페이지에 이미지를 띄우도록 구현

Ojeegu committed:
docs : Update README.md file

README 파일 업데이트

pbg0205 commented on src/main/java/kr/codesqaud/cafe/domain/Article.java:
```diff
@@ -0,0 +1,98 @@
+    private String title;
+    private String content;
+    private String userId;
+    private String create_Time;
```
필드명을 카멜 케이스 사용을 변경해주세요~

Ojeegu replied:
✔ create_Time → createTime으로 바꾸겠습니다~

pbg0205 commented on src/main/java/kr/codesqaud/cafe/controller/dto/ArticleDTO.java:
```diff
@@ -0,0 +1,48 @@
+    private final String title;
+    private final String content;
+    private final String userId;
+    private final String create_Time;
```
카멜 케이스로 변경해주세요~

Ojeegu replied:
✔ create_Time → createTime으로 바꾸겠습니다~

pbg0205 commented on src/main/java/kr/codesqaud/cafe/service/UserService.java:
```diff
@@ -0,0 +1,84 @@
+    }
+
+    private long obtainId(HttpSession session) {
+        LoginDTO loggedInUser = (LoginDTO) session.getAttribute("loginUser");
```
매직 스트링은 상수 또는 변수로 변경해주세요~

Ojeegu replied:
🧐 loginUser를 `private static final String LOGIN_USER_ATTRIBUTE = "loginUser"` 이런 식으로 빼면 될까요?
그리고 매직 스트링 사용을 지양하는 이유는 유지보수가 어렵기 때문이 맞을까요?

Ojeegu replied:
🤔 loginUser가 `LoginController`와 `UserService, ArticleService`에서 중복으로 사용되어서 `LoginController`에서 public static 상수로 선언하여 사용하도록 수정했는데, public으로 선언해도 괜찮을까요? 각 클래스 마다 상수로 빼는 것은 비효율적인 것 같아서요.

pbg0205 replied:
`pubilc static` 으로 관리해야 한다면 별도의 상수 클래스를 통해 별도로 상수를 관리하는 방법이 더욱 적합한 것 같아요 😃

pbg0205 commented on src/main/java/kr/codesqaud/cafe/service/UserService.java:
```diff
@@ -0,0 +1,84 @@
+
+    public ProfileDTO findOne(final long id) {
+        Optional<User> wantedUser = userRepository.findById(id);
+        return wantedUser.map(ProfileDTO::from).orElse(null);
```
원한느 데이터가 존재하지 않을 경우, null 이 넘어가면 이슈가 생길 것 같아서 예외 처리를 추가해주시면 좋을 것 같습니다 😃

Ojeegu replied:
📝 네, 다음주 중으로 예외처리 해보겠습니다~

pbg0205 commented on src/main/java/kr/codesqaud/cafe/controller/UserController.java:
```diff
@@ -0,0 +1,71 @@
+    //todo : MvcConfigure 활성화 하려면 get, post url 달라야 함. 변경하기
+    @PostMapping("/users/new-try")
+    public String signUp(@ModelAttribute final JoinDTO joinDTO, Model model) {
+        boolean isIdDuplicated = userService.checkDuplicate(joinDTO.getUserId());
```
내부 로직에서는 중복을 확인한다는 느낌보다는 유저가 존재하는지를 확인한다는 의미로 existUser 와 같은 네이밍은 어떠세요??

Ojeegu replied:
🙄 제가 boolean 타입에는 is를 붙이는 습관이 있어서요...ㅎㅎ 혹시 `isExistUser`으로 고쳤는데 이것도 괜찮을까요?

pbg0205 replied:
일반적으로는 변수명은 명사 형태로 작성하는 것을 권장하지만 저는 의미를 명확히 전달할 수 있다며 동사도 괜찮지 않을까 생각합니다!! (다른 분들은 다르게 생각할 수 있어서 협업하실 때나 회사가서 작업하실 경우에는 팀 규칙대로 따르시면 될 것 같아요ㅎㅎ)

pbg0205 commented on src/main/java/kr/codesqaud/cafe/controller/UserController.java:
```diff
@@ -0,0 +1,71 @@
+        return "user/modifyProfile";
+    }
+
+    @PutMapping("/users/{id}/revision-try")
```
PUT 은 수정의 의미를 가지고 있어서 `revision-try` 는 제거해도 되지 않을까 싶어요~~

pbg0205 commented on src/main/java/kr/codesqaud/cafe/domain/Article.java:
```diff
@@ -0,0 +1,98 @@
+//
+//    public String getCreate_Time() {
+//        return create_Time;
+//    }
```
불필요한 주석은 제거해주세요 😃

pbg0205 commented on src/main/java/kr/codesqaud/cafe/controller/LoginController.java:
```diff
@@ -0,0 +1,50 @@
+    public String logout(HttpServletRequest request) {
+        HttpSession session = request.getSession(false); // 현재 세션이 없으면 null 반환
+        if (session != null) {
+            session.invalidate(); // 현재 세션을 무효화
```
세션 만료👍👍

pbg0205 commented on src/main/java/kr/codesqaud/cafe/controller/LoginController.java:
```diff
@@ -0,0 +1,50 @@
+        return "redirect:/";
+    }
+
+    @PostMapping("/logout-try")
```
try 는 제거하셔도 될 것 같아요~

pbg0205 commented on src/main/java/kr/codesqaud/cafe/controller/LoginController.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    //todo : MvcConfigure 활성화 하려면 get, post url 달라야 함. 변경하기
+    @PostMapping("/login-try")
```
try 제거하셔도 될 것 같습니다!

pbg0205 commented on src/main/java/kr/codesqaud/cafe/repository/JdbcTemplateUserRepository.java:
```diff
@@ -0,0 +1,76 @@
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.util.*;
```
어떤 패키지를 사용했는지 명확히 알 수 있도록 single import 로 변경해주시면 좋을 것 같습니다 😃

Ojeegu replied:
✔ 수정했습니다. single import를 사용하면 (1) 코드를 읽는 사람이 어떤 클래스가 사용되는지 파악하기 쉽다는 것과 (2) '*'로 처리하는 경우보다 컴파일 시간이 빠르다는 장점이 있다는 것을 알게 되었습니다😃

pbg0205 commented on src/main/java/kr/codesqaud/cafe/service/LoginService.java:
```diff
@@ -0,0 +1,33 @@
+//    public boolean checkLoginUser(LoginDTO loginDTO) {
+//        Optional<User> loginUser = userRepository.findByUserId(loginDTO.getUserId());
+//        return loginUser.isPresent() && loginUser.get().getPassword().equals(loginDTO.getPassword());
+//    }
```
불필요한 주석은 제거해주세요~

pbg0205 commented on src/main/java/kr/codesqaud/cafe/service/LoginService.java:
```diff
@@ -0,0 +1,33 @@
+//    }
+
+    public boolean checkLoginUser(LoginDTO loginDTO) {
+        Optional<User> loginUser = userRepository.findByUserId(loginDTO.getUserId());
```
만약 유저가 존재하지 않을 경우에 예외츨 처리하는 방법으로도 고민해보시면 좋을 것같습니다😃

Ojeegu replied:
📝 네, 다음주에는 예외 처리 해보겠습니다~

pbg0205 commented on src/main/java/kr/codesqaud/cafe/service/UserService.java:
```diff
@@ -0,0 +1,84 @@
+        을 한줄로 변환하면
+        => return wantedUser.map(UserReadDTO::toUserReadDTO).orElse(null);
+
+ */
```
불필요한 주석 제거해주세요~

pbg0205 commented on src/main/java/kr/codesqaud/cafe/unit/CurrentDateTime.java:
```diff
@@ -0,0 +1,17 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class CurrentDateTime {
```
해당 객체의 책임이 문자열로 보여주기 위한 목적인 것 같아요~ 문자열로 보여주기 위한 부분이라면 해결하는 방법이 다양하게 존재하는데 domain 로직에서 DTO 로 넘기는 경우에 지구가 작업해주신 것과 같이 알맞는 포맷으로 변경하는 방법이 있고 thymeleaf 에서 제공하는 경우도 존재합니다.한번 두가지 방법 중에 지구가 더욱 괜찮다고 생각이 드는 방법으로 선택해서 적용해 보시고 혹시 어려움 있으시면 말씀해주세요! (아래 관련 포스팅 첨부드릴게요😃)

- https://memo-the-day.tistory.com/141

Ojeegu replied:
✔ Thymeleaf가 제공하는 기능을 사용하니 훨씬 깔끔하고 유지보수도 쉬워진 것 같습니다~

pbg0205 commented on src/main/resources/schema.sql:
```diff
@@ -0,0 +1,18 @@
+    ID         BIGINT PRIMARY KEY AUTO_INCREMENT,
+    TITLE      VARCHAR(100),
+    CONTENT    LONGTEXT,
+    USERID     VARCHAR(50),
```
외래키에 관한 제약사항에 대해서도 고민해주시면 좋을 것 같습니다~ 외래키를 사용하는 이유는 무엇일까요?

Ojeegu replied:
🙄 외래키는 관계형 데이터베이스에서 2개 이상의 테이블을 연결할 때 사용하는 것으로 알고 있습니다. 현재 외래키를 따로 설정하지 않고 로직으로만 User table의 userId와 Article table의 userId를 연관시켰는데, 이 방법 보다는 외래키를 설정하는 것이 더 효율적일까요?

pbg0205 replied:
외래키를 제약 조건을 선언하는 방법이 무조건 좋다고는 말씀드릴 수는 없을 것 같아요~ 외래키 제약 조건을 말씀드린 의도는 학습해보시고 내부적으로 어떻게 동작하는지 관련해서 학습해보셨으면 하는 의도로 전달드렸어요ㅎㅎ 외래키 제약 조건에 관련되서 한번 학습 해보시고 이후에 같이 어떤 방법이 더 좋을지 기준을 만들어 가보는 것도 좋을 것 같습니다 😃

pbg0205 reviewed (changes requested):
안녕하세요 지구 ~ 리뷰어 쿠퍼입니다!
이번주도 열심히 작업해주셨네요👍👍 수고 많으셨습니다.
아래 문의주신 내용에 관해 답변드리고 추가적으로 궁금한 사항 있으면 코멘트 또는 DM 주시면 답변드리겠습니다ㅎㅎ
(아! 추가적으로 코멘트를 드리면 URL에 try 를 추가하시거나 불필요한 주석을 제거하지 않으시는 것 같아요ㅠ 깔끔한 코드를 작성하기 위해서 미사용 코드 또는 주석을 한번 확인해주시면 더욱 깔끔한 코드가 될 것 같습니다 🫡 ) 

### 1. setter 존재 유무에 따른 뷰가 보이지 않는 현상

- 해당 부분은 NamedParameterJdbcTemplate 를 사용할 때 필요한 요소들에 대해서 학습해보시면 충분히 해결하실 수 있을 거에요😃

### 2. toEntity vs fromEntity

- 제가 말씀드린 의도는 요구사항 변경에 대한 유연성이 있을까 하는 부분에 관해 말씀 드렸어요~ 현재는 단일 도메인을 통해서 DTO 를 만들고 있지만 다양한 요구사항을 만족하기 위해서는 toEntity 방식이 좋을지 fromEntity 방식이 더욱 좋을지에 관해서 고민해보셨으면 해서 전달드렸습니다 😄 한번 고민해보시고 안 떠오르시면 다음에 제 견해에 관해 말씀드릴게요!

### 3. URL 에 관한 내용

- 물론입니다! URL 은 고유한 리소를 가지고 있는 부분이 맞지만 관련된 행동은 HTTP Method 로 정의하는 방식을 권장하고 있어요! 하지만 모든 자원의 행동을 HTTP Method 만으로 만족할 수 없어서 CRUD 외에 다른 요구사항의 경우에는 필요에 따라 URL에 첨부하는 경우도 있습니다.

Ojeegu committed:
refactor : (피드백) 카멜케이스 적용

언더바(_)를 삭제하고 카멜케이스에 맞게 변수명 수정

Ojeegu committed:
refactor : (피드백) 불필요한 주석 삭제

주석 처리하여 사용하지 않는 메서드 삭제

Ojeegu committed:
refactor : (피드백) Magic String을 상수로 선언하여 사용

중복되어 사용되는 Magic String을 static 상수로 선언하여 사용하도록 수정

Ojeegu committed:
refactor : (피드백) 변수명 변경

중복을 확인하기 보다 유저가 존재하는지 확인하는 의미로 변수명을 isDuplicated -> isExistUser 로 변경

Ojeegu committed:
refactor : (피드백) 불필요한 주석 삭제

참고용으로 작성해놓은 주석 불필요하므로 삭제

Ojeegu committed:
refactor : (피드백) Thymeleaf에서 제공하는 날짜 생성 기능 사용

util package의 날짜 변환 class를 사용하는 대신 Thymeleaf에서 제공하는 날짜 생성 및 문자열 변환 기능을 사용하여 DTO로 createTime을 전달하도록 수정

Ojeegu committed:
refactor : (피드백) CurrentDateTime class 삭제

Thymeleaf 기능 사용으로 CurrentDateTime class 기능이 불필요해졌으므로 삭제

Ojeegu committed:
refactor : (피드백) single import로 변경

가독성 및 컴파일 속도 향상을 위해 single import로 변경

Ojeegu committed:
refactor : 데이터베이스 컬럼명 변경 적용 

이전에 컬럼명 변경했던 것을 적용하지 않아서 오류 발생. 현재 사용하는 컬럼명으로 수정.

Ojeegu committed:
refactor : (피드백) URL 경로 수정

URL 경로를 RESTful 성격에 맞게 Http method와 결합하여 적절하게 수정

pbg0205 reviewed (approved):
수고 많으셨습니다 지구! 많은 고민들을 통해 좋은 방법에 관해 고민하고 계셔서 더욱 코드가 좋아지고 있는 것 같아요~ 깐깐하게 코드 리뷰를 하는 것 같다고 생각이 들기도 하는데 항상 긍정적으로 받아드려 주셔서 감사합니다 🙇 이번주는 이정도만 반영해도 괜찮을 것 같아서 추가적으로 개선할 부분은 차주에 반영해주시면 될 것 같습니다. 이번주도 학습 하시느라 수고 많으셨습니다~ 편안한 방 보내시고 다음주에 뵐게요 ✋

pbg0205 commented:
해당 브랜치 remote repo main 브랜치를 바라보고 있어서 브랜치 변경 이후 꼭 전달해주세요!!! 만약의 경우를 대비해서 우선 PR close 처리해둘게요
